src/esrs/state.rs

```diff
@@ -10,48 +11,49 @@ use crate::types::SequenceNumber;
 /// - a lock representing the atomicity of the access to the aggregate,
 /// - a state defined by the user of this library.
 pub struct AggregateState<S> {
-    pub(crate) id: Uuid,
-    pub(crate) sequence_number: SequenceNumber,
-    pub(crate) lock: Option<EventStoreLockGuard>,
-    pub(crate) inner: S,
-}
-
-impl<S: Clone> Clone for AggregateState<S> {
-    fn clone(&self) -> Self {
-        Self {
-            id: self.id,
-            sequence_number: self.sequence_number,
-            inner: self.inner.clone(),
-            lock: None,
-        }
-    }
+    id: Uuid,
+    sequence_number: SequenceNumber,
+    lock: Option<EventStoreLockGuard>,
+    inner: S,
 }
 
 impl<S: std::fmt::Debug> std::fmt::Debug for AggregateState<S> {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
         f.debug_struct("AggregateState")
             .field("id", &self.id)
             .field("sequence_number", &self.sequence_number)
-            .field("inner", &self.inner)
             .field("lock", &self.lock.is_some())
+            .field("inner", &self.inner)
             .finish()
     }
 }
 
 /// Default implementation for [`AggregateState`]
 impl<S: Default> Default for AggregateState<S> {
     fn default() -> Self {
-        Self::new(Uuid::new_v4())
+        Self::new()
     }
 }
 
 impl<S: Default> AggregateState<S> {
-    /// Creates a new instance of an [`AggregateState`] with the given aggregate id. The use of this
-    /// is discouraged being that that aggregate id could be already existing and a clash of ids
-    /// might happen.
+    /// Creates a new instance of an [`AggregateState`] with a new unique id.
+    pub fn new() -> Self {
```

Comment:
Trying to be idiomatic, I've left the standard `new` constructor that uses a random id.
`AggregateState::default` invokes this as well, I've left it because it doesn't hurt but I prefer an explicit constructor.